bot/seasons/halloween/halloween_facts.py

```diff
@@ -30,37 +32,25 @@ def __init__(self, bot):
         with open(Path("bot", "resources", "halloween", "halloween_facts.json"), "r") as file:
             self.halloween_facts = json.load(file)
         self.channel = None
-        self.last_fact = None
+        self.facts = list(enumerate(self.halloween_facts))
+        random.shuffle(self.facts)
 
     async def on_ready(self):
-        self.channel = self.bot.get_channel(HACKTOBER_CHANNEL_ID)
+        self.channel = self.bot.get_channel(Hacktoberfest.channel_id)
         self.bot.loop.create_task(self._fact_publisher_task())
 
-    async def _fact_publisher_task(self):
-        """
-        A background task that runs forever, sending Halloween facts at random to the Discord channel with id equal to
-        HACKTOBER_CHANNEL_ID every INTERVAL seconds.
-        """
-        facts = list(enumerate(self.halloween_facts))
-        while True:
-            # Avoid choosing each fact at random to reduce chances of facts being reposted soon.
-            random.shuffle(facts)
-            for index, fact in facts:
-                embed = self._build_embed(index, fact)
-                await self.channel.send("Your regular serving of random Halloween facts", embed=embed)
-                self.last_fact = (index, fact)
-                await asyncio.sleep(INTERVAL)
+    @property
```

Comment:
This being a property feels weird, any specific reason why it is?

Reply:
I did not write this cooode. I guess I'll fix it if this is gonna be a review of all the hacktoberbot PRs.